### ISSUE TITLE
CI cannot detect a rejected .kicad_dru - kicad-cli reports it as a clean run

### ISSUE THREAD

**Issue by Cimos** [Cimos/KiCad-CustomDesignRules#35]
Nothing in CI can tell whether a `.kicad_dru` was accepted. A single bad layer name makes KiCad discard the **whole file**, and `kicad-cli pcb drc` reports that as a clean run.

Two compounding facts, both measured on KiCad 9.0.6 and 10.0.5:

**The blast radius is the file, not the rule.** Put a known-good rule next to one with an unresolvable layer name and the good rule stops firing too. Every rule reverts to KiCad defaults.

| Rule file contents | violations |
|---|---|
| good rule alone | 51 |
| good rule + rule with bad layer name | 30 (baseline — both dead) |
| no custom rules at all | 30 |

**The CLI says nothing.** No stdout message, empty stderr, exit code 0. The `Unrecognized layer` error exists only in the PCB editor GUI, which is why #33 was reported by a user and not by us.

```mermaid
flowchart TD
    A["one bad layer name"] --> B["KiCad rejects the entire .kicad_dru"]
    B --> C["PCB editor: Unrecognized layer error"]
    B --> D["kicad-cli pcb drc: exit 0, empty stderr"]
    D --> E["CI green, rules never ran"]
```

`tools/lint_dru.py` does not close the gap either — it has no layer vocabulary, so it passes both rule files today despite #33. The `drc.yml` workflow proposed in #32 would not catch it, since it is informational and omits `--exit-code-violations`.

Worth fixing at the linter, since the CLI cannot be relied on here.

<details>
<summary>Why a wrong layer name fails to resolve</summary>

A `.kicad_dru` layer name is matched against a table holding the file-format name (`F.SilkS`, always present) plus the board's current layer name, which is `F.Silkscreen` only while the board keeps KiCad's default. Boards imported from Altium, Eagle, EasyEDA, CADSTAR or Fabmaster carry the source tool's naming, and any `.kicad_pcb` older than file version 20200922 (KiCad 5.1 and earlier) loads with the file-format name showing.
</details>

**Comment by Cimos** [comment]
Addressed in **[kicad-druid](https://github.com/Cimos/kicad-druid)**, which this project has moved to.

`kicad-cli pcb drc` still reports a rejected rules file as a clean run — that is KiCad's behaviour and not something this repo can change. What changed is that the gap is now covered where it can be: `tools/lint_dru.py` validates every layer name in `(layer ...)` clauses, `.Layer` comparisons and `existsOnLayer()` calls against the file-format vocabulary, handling the `?`/`*` wildcards and the `outer`/`inner` keywords. Display-name spellings get a message naming the correct form; unknown layers are rejected outright.

The README also warns that a green DRC run does not prove your custom rules ran, and tells people to check `Board Setup > Design Rules > Custom Rules` after adding a file.

**Comment by Cimos** [comment]
Addressed in kicad-druid — the linter validates every layer name (kicad-cli itself still reports a rejected rules file as a clean run; that is KiCad behaviour). The README there documents the trap.
